lib/iris/_shapefiles.py

```
@@ -3,241 +3,484 @@
+    if not isinstance(geometry, sgeom.MultiPoint):
+        if bool(abs(geometry.bounds[2] - geometry.bounds[0]) > 180.0):
+            antimeridian_warning_msg = (
+                "Geometry crossing the antimeridian is not supported. "
+                "Cannot verify non-crossing given current geometry bounds."
+            )
+            warnings.warn(antimeridian_warning_msg, category=IrisUserWarning)
+
+    # Check if the geometry crosses the poles
+    npole = sgeom.Point(0, 90)
+    spole = sgeom.Point(0, -90)
+    if geometry.intersects(npole) or geometry.intersects(spole):
+        err_msg = "Geometry crossing the poles is not supported."
+        raise ValueError(err_msg)
+
+    return
+
+
+def _transform_geometry(
+    geometry: shapely.Geometry, geometry_crs: ccrs.CRS | CRS, cube_crs: ccrs.CRS
+) -> shapely.Geometry:
+    """Transform a geometry to the cube CRS using pyproj.
 
     Parameters
     ----------
-    cube : :class:`iris.cube.Cube`
+    geometry : :class:`shapely.Geometry`
+        The geometry to transform.
+    geometry_crs : :class:`cartopy.crs`, :class:`pyproj.CRS`
+        The coordinate reference system of the geometry.
+    cube_crs : :class:`cartopy.crs`, :class:`pyproj.CRS`
+        The coordinate reference system of the cube.
 
     Returns
     -------
-    tuple of str
-        The names of the primary latitude and longitude coordinates.
-
+    :class:`shapely.Geometry`
+        The transformed geometry.
     """
-    latc = (
-        cube.coords(axis="y", dim_coords=True)[0]
-        if cube.coords(axis="y", dim_coords=True)
-        else -1
-    )
-    lonc = (
-        cube.coords(axis="x", dim_coords=True)[0]
-        if cube.coords(axis="x", dim_coords=True)
-        else -1
-    )
+    # Set-up transform via pyproj
+    t = Transformer.from_crs(
+        crs_from=geometry_crs, crs_to=cube_crs, always_xy=True
+    ).transform
+    # Transform geometry
+    return shapely.ops.transform(t, geometry)
```

Need to use the PyProj `errcheck=True` argument. This will raise informative errors instead of returning `Inf`.

This should replace the current disabling of PROJ networking, which in some cases would cause unnecessary errors - when network resources are needed and available. The most common cause of networking errors is an incorrect SSL certificate path; we need to make users aware so that they can fix (`pyproj.network.set_ca_bundle_path()`).

I'm hoping and assuming this is our only call to PROJ or PyProj? There's nothing within our Rasterio call? That would get more difficult.

---

Edgecase unit test now added as well.